frontend/apps/reader/readerui.lua

```diff
@@ -382,6 +382,10 @@ function ReaderUI:init()
         })
     end
 
+    -- Allow others to change settings based on external factors
+    UIManager:broadcastEvent(Event:new("DocSettingsLoad", self.doc_settings, self.document.file))
+    self.docsettingtweak:handleEvent(Event:new("DocSettingsLoad", self.doc_settings, self.document.file))
```

Comment:
Or you could just do as on the next line (if the next line work, this should):
```lua
+    -- Allow others to change settings based on external factors, before feeding it to onReadSettings
+    self:handleEvent(Event:new("TweakSettings", self.doc_settings))
+
     -- we only read settings after all the widgets are initialized
     self:handleEvent(Event:new("ReadSettings", self.doc_settings))
```
Also, because plugins can be disabled, your `self.docsettingtweak:handleEvent` would cause a crash when `self.docsettingtweak` does not exist.